patches/server/0018-Multithreaded-entity-tracking.patch

```diff
@@ -254,17 +254,21 @@ index 526c1419af7bd0b6098a8f9a0a24a64ba61c6ebc..864d5a0bd55fd763e51759bc5a5ce951
  
          // CraftBukkit start - Fix for nonsensical head yaw
 diff --git a/src/main/java/net/minecraft/server/PlayerChunkMap.java b/src/main/java/net/minecraft/server/PlayerChunkMap.java
-index a42571cfd2c9c80df27e59db832cb64c2a64e141..e265a10781286a94d81086731ec0d3330dc3977f 100644
+index a42571cfd2c9c80df27e59db832cb64c2a64e141..c0c8d6da17e2e1e89e6842eaab9662b1c64e7b6d 100644
 --- a/src/main/java/net/minecraft/server/PlayerChunkMap.java
 +++ b/src/main/java/net/minecraft/server/PlayerChunkMap.java
-@@ -2119,10 +2119,30 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
+@@ -2119,10 +2119,34 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
          entity.tracker = null; // Paper - We're no longer tracked
      }
  
 +    // Airplane start - tools to ensure main thread
 +    private final java.util.concurrent.ConcurrentLinkedQueue<Runnable> trackerMainQueue = new java.util.concurrent.ConcurrentLinkedQueue<>();
 +    private gg.airplane.structs.TrackQueue trackQueue;
 +
++    PlayerChunk trackerGetVisibleChunk(long key) {
++        return ((ProtectedVisibleChunksMap)this.visibleChunks).safeGet(key);
```

Comment:
This should be fine looking at the code, assuming the distance manager never gets ticked while the entity tracker updates. Let me know if you see anything that can be an issue, although maybe we should just add an exception if the  map gets updated during the entity tracker update.

Reply:
Distance manager is protected by async catcher so it should be fine.

Reply:
So I got readonly protection now.